OasisDEX/oasis-borrow
# Prevent uBlock from blocking mixpanel analytics

# [Prevent uBlock from blocking mixpanel analytics](https://app.shortcut.com/oazo-apps/story/5645/prevent-ublock-from-blocking-mixpanel-analytics)


  
## Changes 👷‍♀️
- remove double tracking (directly from FE + proxied) for FE events
- proxy all tracked events through mixpanel endpoint (`/api/t`)
- added `$current_url` , `$initial_referrer` and `$initial_referring_domain` as per `https://help.mixpanel.com/hc/en-us/articles/115004613766-Default-Properties-Collected-by-Mixpanel` to supplement the data forwarded by proxy
- moved ` mixpanelInit()` in `_app.tsx` to the `useEffect` so `window` object  is visible ( we can get current URL)
-  removed `be-` prefix to keep the current data format
- if the user opted out (cookie settings) track only `AccountChange` address and `ConfirmTransaction`
  
## How to test 🧪

- change `MIXPANEL_ENV` to `production`
- remove `if (process.env.NODE_ENV !== 'development') {` in `_app.tsx` to track route changes
- check if the calls to `/api/t` are successful
- check events mixpanel dashboard in `Testing Environments`
- check mixpanel events in terminal when `MIXPANEL_ENV=='developement'`
